src/aem/azext_aem/custom.py

```diff
@@ -0,0 +1,546 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from knack.util import CLIError
+from knack.log import get_logger
+
+from azure.cli.core.profiles import ResourceType, get_sdk
+from azure.cli.core.commands.client_factory import get_mgmt_service_client, get_data_service_client
+
+logger = get_logger(__name__)
+
+aem_extension_info = {
+    'Linux': {
+        'publisher': 'Microsoft.OSTCExtensions',
+        'name': 'AzureEnhancedMonitorForLinux',
+        'version': '3.0'
+    },
+    'Windows': {
+        'publisher': 'Microsoft.AzureCAT.AzureEnhancedMonitoring',
+        'name': 'AzureCATExtensionHandler',
+        'version': '2.2'
+    }
+}
+
+
+def set_aem(cmd, resource_group_name, vm_name, skip_storage_analytics=False):
+    aem = EnhancedMonitoring(cmd, resource_group_name, vm_name,
+                             vm_client=get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_COMPUTE),
+                             storage_client=get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_STORAGE),
+                             skip_storage_analytics=skip_storage_analytics)
+    aem.enable()
+
+
+def delete_aem(cmd, resource_group_name, vm_name):
+    aem = EnhancedMonitoring(cmd, resource_group_name, vm_name,
+                             vm_client=get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_COMPUTE),
+                             storage_client=None)
+    aem.delete()
+
+
+def verify_aem(cmd, resource_group_name, vm_name, wait_time_in_minutes=15, skip_storage_check=False):
+    aem = EnhancedMonitoring(cmd, resource_group_name, vm_name,
+                             vm_client=get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_COMPUTE),
+                             storage_client=get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_STORAGE))
+    aem.verify(skip_storage_check, wait_time_in_minutes)
```

Comment:
Should any of the commands return the output from `aem.verify/enable/delete`?
Right now, looks like the commands wouldn't return anything.

Reply:
I made it like `all good unless you see an error emitted`, but let me know your reference

Reply:
That's fine.